javascript/ql/src/Security/CWE-400/PrototypePollutionUtility.ql

```diff
@@ -154,6 +161,72 @@ class DynamicPropRead extends DataFlow::SourceNode, DataFlow::ValueNode {
   }
 }
 
+/**
+ * Holds if `output` is the result of `base[key]`, either directly or through
+ * one or more function calls.
+ */
+predicate dynamicPropReadStep(Node base, Node key, SourceNode output) {
+  exists(DynamicPropRead read |
+    not read.hasDominatingAssignment() and
```

Comment:
Can you motivate in a comment why we need this conjunct. It is not clear from the docstring that it is needed. 
I find the need for the two other uses of `hasDominatingAssignment` in this file clearer since they are in `isAdditionalFlowStep`.